sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -172,27 +178,39 @@ a!
 ----
 namespace sycl::ext::oneapi::experimental {
 
+template<typename PropertyListT = empty_properties_t>
 kernel_bundle<bundle_state::ext_oneapi_source> create_kernel_bundle_from_source(
   const context& ctxt,
   source_language lang,
-  const std::string& source)
+  const std::string& source,
+  PropertyListT props = {})
 
+template<typename PropertyListT = empty_properties_t>
 kernel_bundle<bundle_state::ext_oneapi_source> create_kernel_bundle_from_source(
   const context& ctxt,
   source_language lang,
-  const std::vector<std::byte>& bytes)
+  const std::vector<std::byte>& bytes,
+  PropertyListT props = {})
 
 } // namespace sycl::ext::oneapi::experimental
 ----
 !====
 
+_Constraints:_ Available only when `PropertyListT` is an instance of
+`sycl::ext::oneapi::experimental::properties` which contains no properties
+other than those listed below in the section "New properties for the
+`create_kernel_bundle_from_source` function".
+
 _Preconditions:_ There are two overloads of this function: one that reads the
 source code of the kernel from an `std::string`, and one that reads the source
 code of the kernel from an `std::vector` of `std::byte`.
 Each source language `lang` specifies whether the language is text format or
 binary format, and the application must use the overload that corresponds to
 that format.
```

Comment:
> Is this a _Precondition_, i.e., does using an invalid source language and format result in undefined behavior? Maybe we should move this under _Throws_ instead?

Good idea.  I added a _Throws_ section in ce3bbbed6556500e98c689ec3368f958994bc397.

> Should we maybe introduce an aspect to check a device supports a given source language?

We don't need an aspect because `device::ext_oneapi_can_compile` provides this query.